examples/preconditioned-solver/preconditioned-solver.cpp

```diff
@@ -86,27 +87,35 @@ int main(int argc, char* argv[])
 
     // executor where Ginkgo will perform the computation
     const auto exec = exec_map.at(executor_string)();  // throws if not valid
+    exec->add_logger(gko::log::create_roctx_hook());
```

Comment:
Since all of these loggers need to be run under a specific wrapper (`ncu`, `rocprof` or `tau`), do you think this would be directly useful?

Reply:
I think so. If someone wants to create traces both on amd and nvidia systems, it would be annoying to switch back and forth in the code. And there seems to be only one option for hic/cuda respectively, so it makes sense to just pick that one. Perhaps on cpu there is some intel-based thing or other solutions, but I would also be fine with restricting our selves to tau.